### ISSUE TITLE
"mongocli atlas networking peering list" return none where there is peering

### ISSUE THREAD

**Issue by SteveH-US** [mongodb/mongocli#954]
**Describe the bug**
A clear and concise description of the bug.

I can see peering in the Atlas WebUI, but the CLI does not return the data
```
mongocli atlas networking peering list --projectId 61af7800d97aa745bf62691a
[]
```

**To Reproduce**
1. Run 'mongocli atlas networking peering list ` on project/cointainer that has peering configured
2. See no peering returned

**Expected behavior**
Please explain what you expected to happen.
I expect the CLI to return the peering if it exists. 

**Screenshots**
If possible, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. Ubuntu Linux 18.04]
 - MongoCLI version [e.g. 1.2]
 - Ops Manager Version: [if applicable]

**Additional context**

**Comment by andreaangiolillo** [comment]
Hi @SteveH-US,

Are you sure you are using the correct `projectId`? I am asking because I cannot reproduce the issue : (. 
Which mongocli version are you using? Thanks!


```bash
mongocli atlas networking peering list --projectId 5e4e593f70dfbf1010295836
ID                         STATUS               CONTAINER ID
61f2d6b07430f3707b2f3fa6   PENDING_ACCEPTANCE   61f2d6afded017597e54edc7
```
![Screenshot 2022-01-27 at 17 33 22](https://user-images.githubusercontent.com/5663078/151412514-54d3c60d-945c-4d7d-8b89-4b3e4255c48d.png)

**Comment by SteveH-US** [comment]
I'm using
```shell
$ mongocli --version
mongocli version: 1.20.4
git version: homebrew-release
Go version: go1.17.2
   os: darwin
   arch: amd64
   compiler: gc
```

I get no results regardless of which project I use. I know that this particular one has networking peers because I can see them from the WebUI. 

The API KEY I'm using has "Organization Read Only" access.

**Comment by andreaangiolillo** [comment]
Hi,
I have tried to use mongocli v1.20.4 with `Organization Read Only` keys but I couldn’t reproduce the issue. 

```bash
./mongocli atlas networking peering list --projectId 5e4e593f70dfbf1010295836
ID                         STATUS               CONTAINER ID
61f2d6b07430f3707b2f3fa6   PENDING_ACCEPTANCE   61f2d6afded017597e54edc7

A new version of mongocli is available 'v1.21.0'!
To upgrade, see: https://dochub.mongodb.org/core/mongocli-install.

To disable this alert, run "mongocli config set skip_update_check true".
```
Mongocli is calling this [endpoint](https://docs.atlas.mongodb.com/reference/api/vpc-get-connections-list/) to list the available network peering connections. Do you mind calling this endpoint with curl? You should run something similar to this piece of code (you need to update it with your public/private keys and projectId)

```bash
curl --user "{PUBLIC-KEY}:{PRIVATE-KEY}" --digest \
  --header "Accept: application/json" \
  --request GET "https://cloud.mongodb.com/api/atlas/v1.0/groups/{GROUP-ID}/peers?providerName=AWS&pretty=true"
```

Depending on the response that you get, we may know if it is an issue with Mongocli or with the underline endpoint. Thanks!

**Comment by SteveH-US** [comment]
Hi, 

I tried this RESTful command as recommended on two projects received the same results.

```shell
$ # tried the RESTful command
$ curl --user "{PUBLIC-KEY}:{PRIVATE-KEY}" --digest   --header "Accept: application/json" \
>   --request GET "https://cloud.mongodb.com/api/atlas/v1.0/groups/{GROUP-ID}/peers?providerName=AWS&pretty=true"
{
  "error" : 401,
  "reason" : "Unauthorized",
  "detail" : "You are not authorized for this resource."
}
$
$ # tried its equivalent mongocli command with the same API key and group ID
$ mongocli atlas networking peering list --projectId $GROUP_ID
[]
```

The API that I'm using has "Organization Read Only" privileges. 

Steve

**Comment by gssbzn** [comment]
HI @SteveH-US at this point I think it would be best to open a support case via Atlas support where it be safer to share things like project ID and that would let us look at your particular case to see if this is related to it.

**Comment by gssbzn** [comment]
Also just in case, for `mongocli atlas networking peering list` you need to specify a provider if you dont it defaults to AWS, is your network peering for a different provider?

```bash 
mongocli atlas networking peering list --help
Retrieve details for all network peering connections in one Atlas project.

Usage:
  mongocli atlas networking peering list [flags]

Aliases:
  list, ls

Flags:
  -h, --help               help for list
      --limit int          Number of items per page.
  -o, --output string      Output format.
                           Valid values: json|json-path|go-template|go-template-file
      --page int           Page number.
      --projectId string   Project ID to use. Overrides configuration file or environment variable settings.
      --provider string    Name of your cloud service provider.
                           Valid values: AWS|AZURE|GCP. (default "AWS")
```

Please note that this is API behaviour and not related to the CLI if you are specifying the correct provider I would still recommend you to file a support case to look into this for your particular environment

**Comment by SteveH-US** [comment]
IMHO, the `mongocli` ought to report an "Unauthorized" error rather than a null result. 

There's nothing for support to help with in this case because there is a bug in the `mongocli`. 

Now a support question could be why do I get an "Unauthorized" result from the RESTful API for an API key that has "Organization Read Only" privileges.

**Comment by gssbzn** [comment]
I can understand the frustration at this point @SteveH-US but so far we have no conclusive evidence there's an error with `mongocli`

> mongocli ought to report an "Unauthorized" error rather than a null result.

It does when the API keys are incorrect or have no authorization for the Org or Project you are trying to access.

when running
```bash
curl --user "{PUBLIC-KEY}:{PRIVATE-KEY}" --digest   --header "Accept: application/json" \
   --request GET "https://cloud.mongodb.com/api/atlas/v1.0/groups/{GROUP-ID}/peers?providerName=AWS&pretty=true"
```
did you replace `PUBLIC-KEY` for your public key, `PRIVATE-KEY` for your private key and `GROUP-ID` for your project id? are they correct, do those keys have access to that particular project you're trying to test? do those values match the values you are using on `mongocli`, for example if using the `default` profile doing `mongocli config describe default` matches those used with `curl`?

Can you please share the project ID you are testing, is it still `61af7800d97aa745bf62691a` from your original issue?

Does that project have network peerings?

Do you still not see them when doing `mongocli atlas networking peering list`? 

What cloud provider are those peerings for, as mentioned before both the CLI and the API default to AWS when a provider is not set?

All these questions are better handled by a support case, specially as we'll be discussing information pertaining to your account in a public forum, if there were a bug on the CLI that support case will reach us internally.

On the other hand if you're happy to provide all the information I've asked, starting with project ID that does have network peering, we are happy to provide you with support via a GitHub issue

**Comment by SteveH-US** [comment]
> did you replace PUBLIC-KEY for your public key, PRIVATE-KEY for your private key and GROUP-ID for your project id? are they correct, do those keys have access to that particular project you're trying to test? do those values match the values you are using on mongocli, for example if using the default profile doing mongocli config describe default matches those used with curl?

Yes, in all my examples the API key and project ids used for the `curl` command where the same as those used for `mongocli`.  

> Can you please share the project ID you are testing, is it still 61af7800d97aa745bf62691a from your original issue?
> Does that project have network peerings?
I tried several projects, all having network peerings, to eliminate the case where the failure to return unauthorized was project-specific behavior. 

> Do you still not see them when doing `mongocli atlas networking peering list`?
For all my projects, this command returns nothing. 

> What cloud provider are those peerings for, as mentioned before both the CLI and the API default to AWS when a provider is not set?
GCP

I'm surprised that you are pushing back on what is obviously a bug in `mongocli`. But, I'll submit a support ticket as you've asked. 

BTW, there is another bug in the docs.  You can't use a hyphen in the variable name is bash, e.g.  the "PUBLIC-KEY" is an invalid variable name.

**Comment by gssbzn** [comment]
thanks @SteveH-US, we'll keep an eye for any internal support case for the CLI

And thanks fro confirming you are using GCP, this could be a case where you need to do `mongocli atlas networking peering list --provider GCP` the API the CLI consumes doesn't have the ability to list for all cloud providers 

> the "PUBLIC-KEY" is an invalid variable name.

if this in the public docs for the API I believe this is intentional, I don't think these examples are suppose to be scriptable and intent on users having to replace the values but I'll pass along your feedback
